File: Project-N01/src/main/java/com/toby/HelloWorld.java
package com.toby;

public class HelloWorld {
    // psvm + tab = main function template
    public static void main(String[] args) {
        // sout + tab = println function template

        Person p = new Person("Toby", 80.0f, 1.8f);
        p.hello();
        System.out.println(p.bmi());

        int score = 88;
        System.out.println(score > 80 && score < 90);
        System.out.println(score < 80 || score > 90);

        char c = 'A';
        System.out.println(c > 'a');

        // Basic Variable -> No Method (Lower-case Letter)

        /*int age = 20;
        char c = 'T';
        byte b = 120;
        float weight = 80.0f;
        double height = 1.8;
        boolean adult = true;
        boolean enroll = false;*/

        // Reference Varialbe -> With Method (Capital Letter)

        /*String name = "Toby";*/

        // Wrapper Class

        /*Integer age2 = 20;
        Character c2 = 'T';
        Byte b2 = 120;
        Float weight2 = 80.0f;
        Double height2 = 1.8;
        Boolean adult2 = true;
        Boolean enroll2 = false;*/
    }
}

File: Project-N01/src/main/kotlin/com/toby/HelloWorld.kt
package com.toby

// main + tab = main function template
fun main() {
    // println("Hello Kotlin World!")

    // Human().hello()

    val h = Human(weight = 80.0f, height = 1.8f)
    h.hello()
    println(h.bmi())

    // val = value, not changeable
    // var = variable, changeable

    // If the variable / value is initialize, no need to set type (auto estimation)

    /*val age = 19
    var age2 = 19.0f*/

    // variable / value without initialization, set type with ':'

    /*val age3 : Int
    var age4 : Float
    var name : String = "Toby"*/
}

class Human(var name: String = "", var weight: Float, var height: Float) {
    init {
        println("test $weight $height")
    }

    // constructor(name: String, weight: Float, height: Float) : this(weight, height)

    fun bmi() : Float {
        val bmi = weight / (height * height)
        return bmi
    }

    fun hello() {
        println("Hello Kotlin World!")
    }
}